pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -437,6 +456,22 @@ private BrokerResponse executeSqlQuery(ObjectNode sqlRequestJson, HttpRequesterI
     if (forceUseMultiStage) {
       sqlNodeAndOptions.setExtraOptions(ImmutableMap.of(Request.QueryOptionKey.USE_MULTISTAGE_ENGINE, "true"));
     }
+    if (getCursor != null && getCursor) {
+      if (numRows == null) {
+        numRows = _brokerConf.getProperty(CommonConstants.CursorConfigs.QUERY_RESULT_SIZE,
+            CommonConstants.CursorConfigs.DEFAULT_QUERY_RESULT_SIZE);
+      }
+
+      if (numRows > CommonConstants.CursorConfigs.MAX_QUERY_RESULT_SIZE) {
```

Comment:
Why do we want to introduce the `MAX_QUERY_RESULT_SIZE` here if there is no such restriction on the generic queries?

Reply:
`MAX_CURSOR_NUM_ROWS` is a better name. There is no restriction on the size of the query result. The restriction is on the number of rows in a specific fetch. I added this as a backstop to runaway cursor fetch workloads. Couple of options:
* I can remove it.
* Default to a much larger number. (500K right now).

Preferences?